src/compas/datastructures/mesh/_mesh.py

```diff
@@ -644,11 +643,21 @@ def from_vertices_and_faces(cls, vertices, faces):
 
         """
         mesh = cls()
-        for vkey, vertex in enumerate(vertices):
-            x, y, z = vertex
-            mesh.add_vertex(key=vkey, x=x, y=y, z=z)
-        for fkey, face in enumerate(faces):
-            mesh.add_face(face, fkey=fkey)
+
+        if type(vertices) == list:
```

Comment:
technically, the preferred way of doing this kind of thing would be through *duck typing*. if you are going to check for specific types, the method you used will not always work. something like the following would be better

```python
import collections

# ...

if isinstance(vertices, collections.Mapping):
    # stuff

Reply:
https://docs.python.org/3/library/collections.abc.html

Reply:
great, I used collections.Mapping for dictionaries and collections.MutableSequence for lists